src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -217,6 +217,277 @@ .maxstack 8
+        {
+            var comp = CreateCompilation(@"
+public struct C
+{
+    public static void Main()
+    {
+        var x = new C();
+        x.P = 10;
+        x = new C();
+    }
+
+    public C()
+    {
+        System.Console.WriteLine(""In C..ctor before assignment: "" + P);
+        P = 5;
+        System.Console.WriteLine(""In C..ctor after assignment: "" + P);
+    }
+
+    public int P { get => field; set => field = value; }
+}
+", options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            CompileAndVerify(comp, expectedOutput: @"In C..ctor before assignment: 0
+In C..ctor after assignment: 5
+In C..ctor before assignment: 0
+In C..ctor after assignment: 5").VerifyIL("C..ctor", @"
+{
+  // Code size       73 (0x49)
+  .maxstack  2
+  .locals init (int V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.0
+  IL_0002:  stfld      ""int C.<P>k__BackingField""
+  IL_0007:  ldstr      ""In C..ctor before assignment: ""
+  IL_000c:  ldarg.0
+  IL_000d:  call       ""int C.P.get""
+  IL_0012:  stloc.0
+  IL_0013:  ldloca.s   V_0
+  IL_0015:  call       ""string int.ToString()""
+  IL_001a:  call       ""string string.Concat(string, string)""
+  IL_001f:  call       ""void System.Console.WriteLine(string)""
+  IL_0024:  ldarg.0
+  IL_0025:  ldc.i4.5
+  IL_0026:  call       ""void C.P.set""
+  IL_002b:  ldstr      ""In C..ctor after assignment: ""
+  IL_0030:  ldarg.0
+  IL_0031:  call       ""int C.P.get""
+  IL_0036:  stloc.0
+  IL_0037:  ldloca.s   V_0
+  IL_0039:  call       ""string int.ToString()""
+  IL_003e:  call       ""string string.Concat(string, string)""
+  IL_0043:  call       ""void System.Console.WriteLine(string)""
+  IL_0048:  ret
+}
+").VerifyDiagnostics(
+    // (13,69): warning CS9020: The 'this' object is read before all of its fields have been assigned, causing preceding implicit assignments of 'default' to non-explicitly assigned fields.
+    //         System.Console.WriteLine("In C..ctor before assignment: " + P);
+    Diagnostic(ErrorCode.WRN_UseDefViolationThisSupportedVersion, "P").WithLocation(13, 69)
+    );
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact] // PROTOTYPE(semi-auto-props): Add test with semi-colon setter when mixed scenarios are supported.
+        public void TestPropertyIsReadThenAssignedInStructConstructor_ReadOnlyProperty()
+        {
+            var comp = CreateCompilation(@"
+public struct C
+{
+    public static void Main()
+    {
+        var x = new C();
+        System.Console.Write(x.P);
+    }
+
+    public C()
+    {
+        System.Console.WriteLine(""In C..ctor before assignment: "" + P);
```

Comment:
>System.Console.WriteLine(""In C..ctor before assignment: "" + P); [](http://example.com/codeflow?start=8&length=65)

Now, how is this test different from the previous one?

Reply:
I really do not understand the motivation behind changing this test scenario. We do want to test scenario where constructor contains only assignment.

Reply:
> Now, how is this test different from the previous one?

One has a readonly property, while the other has a property with a setter.

Reply:
> We do want to test scenario where constructor contains only assignment.

Isn't that covered by `TestPropertyIsAssignedInStructConstructor_PropertyHasSetter` and `TestPropertyIsAssignedInStructConstructor_ReadOnlyProperty`?